testing/regress/ecl/parquetPartition.ecl

```diff
@@ -0,0 +1,73 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+//class=parquet
+
+IMPORT Std;
+IMPORT Parquet;
+
+// Define the record layout with explicit field lengths
+datasetRecordLayout := RECORD
+    UNSIGNED4 id;
+    STRING25 name;
+    UNSIGNED4 age;
+    STRING50 city;
+END;
+
+// Create a small dataset - ensure all records have valid data
+smallData := DATASET([
+    {1, 'Alice', 30, 'New York'},
+    {2, 'Bob', 25, 'Los Angeles'},
+    {3, 'Charlie', 40, 'Chicago'}
+], datasetRecordLayout);
+
+// Set options
+overwriteOption := TRUE;
+rowSize := 1024;  // Increased buffer size
+
+// Define base path
+basePath := Std.File.GetDefaultDropZone() + 'regress/parquet/';
+
+// Define partition keys as string arrays
+hivePartitionKey := ['city'];
```

Comment:
This could easily be the cause of your error. The FUNCTIONMACROs for Write expect a string and you are passing an array. The way to pass in multiple keys is like: 'year;month;day'